app/models/bill.rb

```
@@ -7,5 +7,25 @@ class Bill < ApplicationRecord
   has_many :tests, through: :bill_details, source: :billable, source_type: 'Test'
   has_many :doctors, through: :bill_details, source: :billable, source_type: 'Doctor'
   default_scope { where(hospital_id: Hospital.current_id) }
+  def add_medicine(medicine, quantity_added)
+    if medicine.quantity>0 && quantity_added<=medicine.quantity
+      if medicine.update(quantity: medicine.quantity-quantity_added)
+        self.update(price: self.price+=medicine.price*quantity_added)
+        curr_bd=bill_details.find_by(billable: medicine)
+        if curr_bd
+          curr_bd.update(quantity:quantity_added+curr_bd.quantity)
+        else
+          bill_details.create(quantity: quantity_added, billable:medicine, hospital: medicine.hospital) 
+        end       
+      end
+    else
+      flash[:error]= t('medicine.add.failure')
```

You need to return this error to the controller and set the flash message there. Also, in the controller you are using this function in an if condition, but you are not returning a boolean value here.
Do this for `add_medicine` in `purchase_order.rb` as well.